ddtrace/internal/agent.py

```diff
@@ -9,23 +9,42 @@
 from .uds import UDSHTTPConnection
 
 
-DEFAULT_HOSTNAME = "localhost"
-DEFAULT_TRACE_PORT = 8126
+DEFAULT_HTTP_HOSTNAME = "localhost"
+DEFAULT_UNIX_HOSTNAME = "unix"
+DEFAULT_HTTP_TRACE_PORT = 8126
+DEFAULT_UNIX_TRACE_PORT = "///var/run/datadog/dsd.socket"
 DEFAULT_STATS_PORT = 8125
-DEFAULT_TRACE_URL = "http://%s:%s" % (DEFAULT_HOSTNAME, DEFAULT_TRACE_PORT)
+DEFAULT_TRACE_URL = "http://%s:%s" % (DEFAULT_HTTP_HOSTNAME, DEFAULT_HTTP_TRACE_PORT)
 DEFAULT_TIMEOUT = 2.0
+HOST_SET = os.environ.get("DD_AGENT_HOST", os.environ.get("DATADOG_TRACE_AGENT_HOSTNAME")) is not None
+PORT_SET = os.environ.get("DD_AGENT_PORT", os.environ.get("DD_TRACE_AGENT_PORT")) is not None
+DEFAULT_UNIX_DOMAIN_SOCK_AVAIL = os.path.exists("/var/run/datadog/dsd.socket")
 
 ConnectionType = Union[HTTPSConnection, HTTPConnection, UDSHTTPConnection]
 
 
 def get_hostname():
     # type: () -> str
-    return os.environ.get("DD_AGENT_HOST", os.environ.get("DATADOG_TRACE_AGENT_HOSTNAME", DEFAULT_HOSTNAME))
+    hostname = None
+    if HOST_SET or PORT_SET:
```

Comment:
Does this check need to occur here? It is doing the env lookups anyways.

Reply:
I think this is the other option if you don't like those checks. Is this preferable?:
`def get_hostname():
    # type: () -> str
    hostname = os.environ.get("DD_AGENT_HOST", os.environ.get("DATADOG_TRACE_AGENT_HOSTNAME"))
    
    if hostname == None and DEFAULT_UNIX_DOMAIN_SOCK_AVAIL:
        hostname = DEFAULT_UNIX_HOSTNAME
    else:
        hostname = DEFAULT_HTTP_HOSTNAME

    return hostname`

Reply:
Worked with Brett on this and we moved all of the logic to get_trace_url()